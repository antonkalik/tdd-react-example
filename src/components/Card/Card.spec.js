import { render, screen } from "@testing-library/react";
import { Card } from "./index";
import userEvent from "@testing-library/user-event";

describe("Card", () => {
  const cardTitle = "Card Title";
  const cardId = 2;

  it("should render Card", () => {
    render(<Card />);
    expect(screen.getByTestId("card")).toMatchSnapshot();
  });

  it("should render Card with title", () => {
    const { container } = render(<Card title={cardTitle} />);
    const title = container.querySelector("h5");
    expect(title).toHaveTextContent(cardTitle);
  });

  it("should call onRemove function", () => {
    const onRemove = jest.fn();
    render(<Card onRemove={onRemove} id={cardId} />);
    const element = screen.getByRole("button", { name: "Remove" });
    userEvent.click(element);
    expect(onRemove).toHaveBeenCalledWith(cardId);
  });

  it("should render Card without button if onRemove is not passed", () => {
    render(<Card />);
    expect(screen.queryByRole("button")).not.toBeInTheDocument();
  });
});
