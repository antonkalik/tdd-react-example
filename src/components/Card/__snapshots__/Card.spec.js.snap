// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render Card 1`] = `
<div
  class="sc-bczRLJ blGaXB"
  data-testid="card"
/>
`;
